### ISSUE TITLE
resolve auth issue for backups

### ISSUE THREAD

**Issue by immykins** [influxdata/influxdb#16120]
blocker for #15558

Russ is having some issues with using a "master" token to perform a backup from ` jgm-backup-spike`, with errors like `read:authorizations is unauthorized`. He did not have issues with testing when using a newly generated token.

All the context on this issue:

Our backup authorization logic checks to see if the authorization has access to all resource types. If it doesn't, it rejects the request with a 401 response. This makes sense, but becomes a problem if a new resource type is added after a token is created.

Tokens are intentionally immutable, and they are structured to hold a slice of different resource types and whether they have read or write access to them. You can think of it as a whitelist of permissions it can perform actions with. Because of this whitelist approach and its immutability, all existing tokens will be unable to perform backups after a new resource type is added.

We specifically decided that we do *not* want to try to filter out specific resource types based on what the token has access to, and that we think the way that backup auth is designed right now is reasonable.

We discussed adding a separate permission just for backups, but I pushed back against this. It doesn't make sense to me that a user with limited read access could have unlimited read access just because they can create backups.

Two possible approaches I've considered:
- We extend our existing design by allowing wildcards as a resource (e.g. a token can read or write all resource types forever).
- We redesign our auth to be role-based, and have a "master" auth role that has full access to everything forever.

I highly recommend talking to goller or desa before starting on this issue and reach a consensus with them on the right approach to take with this.

Acceptance criteria:
Prevent users from encountering this problem in the future when new resource types are added.
Russ needs to sign off on this once a solution is delivered. He may want us to come up with a way to migrate all existing tokens so that current users aren't blocked from creating backups.

**Comment by russorat** [comment]
@goller might have some ideas about this.

**Comment by kelwang** [comment]
add reference of https://github.com/influxdata/influxdb/issues/15961

**Comment by kelwang** [comment]
as per discussion with @desa, his team will take over with the second suggestion of @jademcgough. Which is planned out for the new year

**Comment by russorat** [comment]
@immykins i know it's been a while, but i just added instructions to help resolve this issue: https://github.com/influxdata/influxdb/issues/18882#issuecomment-663113088

in your case, you will need to add `{"action":"read","resource":{"type":"authorizations"}},{"action":"write","resource":{"type":"authorizations"}}` as well as the other ones probably.

**Comment by lesam** [comment]
Closing as duplicate of https://github.com/influxdata/influxdb/issues/22401

**Comment by lesam** [comment]
I suspect not many people still have operator / admin tokens from 2019 without the `authorizations` resource type, so this is not really a large backup / restore problem currently. The other issue is about the admin token upgrade problem generally, and should fix this problem as well.
